AU2/plugins/custom_plugins/PolicePlugin.py

```diff
@@ -0,0 +1,228 @@
+        return [
+            Dependency(
+                dependentOn="CorePlugin_assassin_pseudonym",
+                htmlComponents=[
+                    AssassinDependentTextEntry(
+                        pseudonym_list_identifier="CorePlugin_assassin_pseudonym",
+                        identifier=self.html_ids["Rank"],
+                        title="Promote police? Must be a number from 0-9",
+                        default=e.pluginState.get(self.identifier, {}).get("rank", {}),
+                    )
+                ]
+            )
+        ]
+
+    def on_event_update(self, e: Event, htmlResponse) -> List[HTMLComponent]:
+        e.pluginState.setdefault(self.identifier, {})["rank"] = htmlResponse[self.html_ids["Rank"]]
+        return [Label("[POLICE] Success!")]
+
+    def on_page_generate(self, htmlResponse) -> List[HTMLComponent]:
+        pass
+        
+        events = list(EVENTS_DATABASE.events.values())
+        events.sort(key=lambda event: event.datetime)
+
+        police_rank_manager = PoliceRankManager()
+        death_manager = DeathManager(perma_death=True)
+        for e in events:
+            police_rank_manager.add_event(e)
+            death_manager.add_event(e)
+
+        police: List[Assassin] = [a for a in ASSASSINS_DATABASE.assassins.values() if a.is_police]
+        dead_police: List[Assassin] = [i for i in police if i.identifier in death_manager.get_dead()]
+        alive_police: List[Assassin] = [i for i in police if i not in dead_police]
+        if GENERIC_STATE_DATABASE.plugin_map.get("WantedPlugin", True):  # i.e. we can separate police into dead and corrupt-dead
+            pass  # TODO after wanted rewrite
+            '''
+            dead_corrupt_police: List[Assassin] = [i for i in dead_police if i.]
+            dead_non_corrupt_police: List[Assassin]
+            '''
+
+        tables = []
+        # Going to assume there are police, because otherwise you wouldn't have this plugin enabled
+        if alive_police:
+            alive_police.sort(key=lambda a: (-int(police_rank_manager.get_rank(a)), a.real_name))
+            rows = []
+            print(GENERIC_STATE_DATABASE.arb_state.get(self.identifier))
+            for a in alive_police:
+                rows.append(
+                    POLICE_TABLE_ROW_TEMPLATE.format(
+                        RANK=GENERIC_STATE_DATABASE.arb_state.get(self.identifier)[police_rank_manager.get_rank(a)],
+                        PSEUDONYM=a.all_pseudonyms(),
+                        NAME=a.real_name,
+                        EMAIL=a.email,
+                        COLLEGE=a.college,
+                        NOTES=a.notes
+                    )
+                )
+            tables.append(
+                POLICE_TABLE_TEMPLATE.format(ROWS="".join(rows))
+            )
+            if dead_police:
+                dead_police.sort(key=lambda a: (-int(police_rank_manager.get_rank(a)), a.real_name))
```

Comment:
nit: I would sort with priority given to `a.college` then `a.real_name`.

Reply:
I don't see college being relevent at all for police, so name seems to make more sense (also why I've removed police addresses from the website)

Reply:
makes sense